ios/brave-ios/Sources/Brave/Frontend/Browser/Search/SearchSuggestionDataSource.swift

```diff
@@ -22,17 +23,30 @@ class SearchSuggestionDataSource {
   // MARK: SearchListSection
 
   enum SearchListSection: Int, CaseIterable {
-    case quickBar
     case searchSuggestionsOptIn
     case searchSuggestions
+    case braveSearchPromotion
     case findInPage
     case openTabsAndHistoryAndBookmarks
-    case aiChat
   }
 
   let tabType: TabType
   let maxSearchSuggestions = 5
   var suggestions = [String]()
+  var siteData: [Site] {
+    set {
+      _siteData = newValue
+    }
+    get {
+      if siteDataFetchLimit == 0 {
+        return _siteData
+      } else {
+        return Array(_siteData.prefix(siteDataFetchLimit))
+      }
+    }
+  }
+  var siteDataFetchLimit = 5
+  private var _siteData = [Site]()
```

Comment:
This is a bit odd, the underlying fetch still grabs all the data but you're adjusting the returned property based on an property. At this point would be better to remove this entirely and do the `prefix` logic in the UI